### PR TITLE
Migrate DBSCAN test from boost to catch2

### PR DESCRIPTION
**Pull request by Abilityguy**
Regarding issue #2523 

The tests ran successfully on my local system build.

### CODE REVIEW

**Comment by Abilityguy** [comment]
Made changes to main_tests/dbscan_test.cpp so that both can be clubbed in the same pr.

File: src/mlpack/tests/dbscan_test.cpp [hunk 1]
```diff
@@ -13,17 +13,16 @@
 #include <mlpack/methods/dbscan/dbscan.hpp>
 #include <mlpack/methods/dbscan/random_point_selection.hpp>
 
-#include <boost/test/unit_test.hpp>
-#include "test_tools.hpp"
+#include "test_catch_tools.hpp"
+#include "catch.hpp"
+
```
Comment (rcurtin) [inline src/mlpack/tests/dbscan_test.cpp:18]:
```suggestion
```
No need for an extra line. :)

File: src/mlpack/tests/dbscan_test.cpp [hunk 2]
```diff
@@ -75,17 +74,18 @@ BOOST_AUTO_TEST_CASE(OutlierTest)
...
-  BOOST_REQUIRE_EQUAL(assignments.n_elem, points.n_cols);
-  BOOST_REQUIRE_EQUAL(assignments[15], SIZE_MAX);
-  BOOST_REQUIRE_EQUAL(assignments[45], SIZE_MAX);
-  BOOST_REQUIRE_EQUAL(assignments[101], SIZE_MAX);
+  REQUIRE(clusters > 0);
+  REQUIRE(assignments.n_elem == points.n_cols);
+  REQUIRE(assignments[15] == SIZE_MAX);
+  REQUIRE(assignments[45] == SIZE_MAX);
+  REQUIRE(assignments[101] == SIZE_MAX);
+
```
Comment (rcurtin) [inline src/mlpack/tests/dbscan_test.cpp:82]:
```suggestion
```

**Review by rcurtin — APPROVED** [review]
Hey @Abilityguy, thanks for taking the time to help with this refactoring!  Everything looks good to me. :+1:  I only have two really minor comments (just some extra lines that we can remove).  The test failures don't appear to be related to this PR in any way, so we can ignore those.

**Review by zoq — COMMENTED** [review]
Thanks for looking into the test transition, no further comments from my side besides the two issues @rcurtin pointed out.

**Comment by Abilityguy** [comment]
Thanks @rcurtin and @zoq. I have committed the required style fixes.
I was looking at the failed test logs and it looked like a memory error.
Is it due to the issue mentioned in #2625 ?

**Comment by zoq** [comment]
> Thanks @rcurtin and @zoq. I have committed the required style fixes.
> I was looking at the failed test logs and it looked like a memory error.
> Is it due to the issue mentioned in #2625 ?

To me it looks like we have to increase the test tolerance.

**Review by zoq — APPROVED** [review]
Looks great, thanks again.

**Pull request merged**